x-pack/apm-server/profiling/collector.go

```diff
@@ -403,24 +403,24 @@ func (e *ElasticCollector) SetFramesForTraces(ctx context.Context,
 
 		err = multiplexCurrentNextIndicesWrite(ctx, e, &elasticsearch.BulkIndexerItem{
 			Index:      StackTraceIndex,
-			Action:     actionIndex,
+			Action:     actionCreate,
 			DocumentID: docID,
 			OnFailure: func(
 				_ context.Context,
 				_ elasticsearch.BulkIndexerItem,
-				resp elasticsearch.BulkIndexerResponseItem,
-				err error,
+				_ elasticsearch.BulkIndexerResponseItem,
+				_ error,
 			) {
-				counterStacktracesFailure.Inc()
-				e.logger.With(
-					logp.Error(err),
-					logp.String("error_type", resp.Error.Type),
-				).Errorf("failed to index stacktrace metadata: %s", resp.Error.Reason)
+				// Error is expected here, as we tried to "create" an existing document.
+				// We increment the metric to understand the origin-to-duplicate ratio.
+				counterStacktracesDuplicate.Inc()
```

Comment:
How accurate should this metric be? There might be other errors - should you be parsing the error reason, and only incrementing the counter for specific errors?

Reply:
```
const docIDAlreadyExists = "version_conflict_engine_exception"
...
if resp.Error.Type == docIDAlreadyExists {
    counterStacktracesDuplicate.Inc()
    return
}
// log other errors here
```

Reply:
You are right, thanks 🙏🏼 
@rockdaboot shared a suggestion that I'm going to implement here.